### DISCUSSION TITLE
pointer and mut question

### DISCUSSION THREAD

**Discussion by k1attila12** [vlang/v#7694] [General]
j:=1111

mut i := &int(&j)

unsafe {
  *i =2222
  }

println(i)
println(typeof(i))
println(j)
println(typeof(j))

in this small example i can modify j even it is immutable.
is it normal or bug ? (i think it is bug)

if i change this line :
   mut i := &int(&j)
to
  mut i := &j

compiler can't compile it (very correctly) because of "j is immutable"

thank you

**Comment by ntrel** [comment]
> mut i := &int(&j)

Casting between pointer types should require `unsafe`, I started working on it but didn't finish it yet.

**Comment by ntrel** [comment]
After [this](https://github.com/vlang/v/pull/7892) is merged, you will see this warning (which will become an error later):

> ptrcast.v:3:11: warning: casting `&int` to `&int` is only allowed in `unsafe` code

Although the types are the same, the cast bypasses mutability checks.
